policyengine_us/variables/gov/irs/income/taxable_income/adjusted_gross_income/above_the_line_deductions/student_loan_interest/student_loan_interest_ald_magi.py

```diff
@@ -0,0 +1,57 @@
+from policyengine_us.model_api import *
+
+
+class student_loan_interest_ald_magi(Variable):
+    value_type = float
+    entity = Person
+    label = "Madofied adjusted gross income for the student loan interest ALD"
+    unit = USD
+    definition_period = YEAR
+    documentation = "Above-the-line deduction for student loan interest"
+    reference = "https://www.law.cornell.edu/uscode/text/26/221"  # (b)(2)(C)
+    defined_for = "student_loan_interest_ald_eligible"
+
+    def formula(person, period, parameters):
+        p_irs = parameters(period).gov.irs
+        not_dependent = ~person("is_tax_unit_dependent", period)
+        gross_income_sources = p_irs.gross_income.sources
+        # Unemployment compensation is not considered in MAGI
+        gross_income = [
+            income_source
+            for income_source in gross_income_sources
+            if income_source != "taxable_unemployment_compensation"
```

Comment:
make this a parameter, as well as the PERSON_ALDS and EXCLUDED_ALDS

Reply:
I am not sure if the same structure as above would work with a list parameter (we tried) 
Looking through the repo, it seems like we have usually created a structure without list parameters when creating such for loops 
e.g. [SNAP](https://github.com/PolicyEngine/policyengine-us/blob/master/policyengine_us/variables/gov/usda/snap/income/deductions/shelter/snap_net_income_pre_shelter.py#L28-L29)
[Income](https://github.com/PolicyEngine/policyengine-us/blob/master/policyengine_us/variables/gov/irs/income/taxable_income/deductions/qualified_business_income_deduction/taxable_income_less_qbid.py#L21-L22)

If the purpose is referencing then the link in the variable reference describes the MAGI composition 

Obviously open to suggestions

Reply:
Those examples exclude themselves to avoid circular references, so I don't see them as policy parameters in the same way as here.

What's the error? When doing something like
```
if income_source not in p.xyz
```
or perhaps something like
```
if income_source not in list(set(p.xyz))
```